### ISSUE TITLE
build() should call clean_vignettes by default if vignettes = TRUE

### ISSUE THREAD

**Issue by krlmlr** [hadley/devtools#810]
## TL;DR
- [ ] new parameter `clean_vignettes = TRUE` to `build()`
- [ ] update docs for `build_vignettes()`, perhaps mention that adding `inst/doc` to `.Rbuildignore` doesn't work
## Details

The documentation of `build_vignette` [(code ref)](https://github.com/hadley/devtools/blob/d2bdbada3f8921978ab13dbcad5897a60cf5f965/R/vignettes.r#L5) mentions this:

> You need to ensure that these files are not included in the built package...

It would be best if `devtools` just did this:

```
build <- function(..., vignettes = TRUE, clean_vignettes = vignettes) {
  if (clean_vignettes)
    clean_vignettes()
  ...
}
```

Or, even better, move the stuff away during the `build` process and put it back when done.

Otherwise, if `build_vignettes` is called just once, all subsequent updates to the original vignette source will be ignored. (If `inst/doc` is in `.Rbuildignore`, vignettes are built but not included in the `.tar.gz`.) Build artifacts, if ignored by the VCS, should have no impact on the result of subsequent builds!

Per https://github.com/hadley/devtools/commit/9bb78f10ce750d2bd24ca655329e3cbf18cb3f37#commitcomment-11326492.

**Comment by hadley** [comment]
Makes sense to me. Want to do a PR?
